packages/rest/src/__tests__/integration/export-openapi-spec.integration.ts

```diff
@@ -0,0 +1,125 @@
+
+describe('exportOpenApiSpec', () => {
+  let app: MyApp;
+  let lastLog = '';
+  let bootCalled = false;
+
+  const log = (formatter: unknown, ...args: unknown[]) => {
+    lastLog = format(formatter, ...args);
+  };
+
+  const expectedSpec = {
+    openapi: '3.0.0',
+    info: {
+      title: 'LoopBack Application',
+      version: '1.0.0',
+    },
+    paths: {
+      '/hello': {
+        get: {
+          'x-controller-name': 'MyController',
+          'x-operation-name': 'hello',
+          tags: ['MyController'],
+          responses: {
+            '200': {
+              description: 'Return value of MyController.hello',
+            },
+          },
+          operationId: 'MyController.hello',
+        },
+      },
+    },
+    servers: [
+      {
+        url: '/',
+      },
+    ],
+  };
+
+  beforeEach(async () => {
+    lastLog = '';
+    bootCalled = false;
+    await givenApp();
+  });
+
+  it('saves the spec to a json file for RestApplication', async () => {
+    const file = path.join(sandbox.path, 'openapi.json');
+    await app.exportOpenApiSpec(file, log);
+    expectJsonSpec(file);
+  });
+
+  it('saves the spec to a json file for RestServer', async () => {
+    const file = path.join(sandbox.path, 'openapi.json');
+    await app.restServer.exportOpenApiSpec(file, log);
+    expectJsonSpec(file);
+  });
+
+  it('writes the spec as a json document to console', async () => {
+    await app.exportOpenApiSpec('-', log);
+    expect(JSON.parse(lastLog)).to.eql(expectedSpec);
+  });
+
+  it('saves the spec to a yaml file', async () => {
+    const file = path.join(sandbox.path, 'openapi.yaml');
+    await app.restServer.exportOpenApiSpec(file, log);
+    expect(lastLog.match(/The OpenAPI spec is saved to .+ openapi\.yaml$/));
+    const content = fs.readFileSync(file, 'utf-8');
+    expect(safeLoad(content)).to.eql(expectedSpec);
+    expect(bootCalled).to.be.true();
+  });
+
+  it('saves the spec to a yml file', async () => {
+    const file = path.join(sandbox.path, 'openapi.yml');
+    await app.restServer.exportOpenApiSpec(file, log);
+    expect(lastLog.match(/The OpenAPI spec is saved to .+ openapi\.yml$/));
+    const content = fs.readFileSync(file, 'utf-8');
+    expect(safeLoad(content)).to.eql(expectedSpec);
+    expect(bootCalled).to.be.true();
+  });
+
+  after(() => sandbox.reset());
+
+  function expectJsonSpec(file: string) {
+    expect(lastLog.match(/The OpenAPI spec is saved to .+ openapi\.json$/));
+    const content = fs.readFileSync(file, 'utf-8');
+    expect(JSON.parse(content)).to.eql(expectedSpec);
+    expect(bootCalled).to.be.true();
+  }
+
+  class MyController {
+    @get('/hello')
+    hello() {
+      return 'Hello';
+    }
+  }
+
+  class MyApp extends RestApplication {
+    constructor(config: ApplicationConfig) {
+      super(config);
+      this.controller(MyController);
+    }
+    async boot(): Promise<void> {
+      bootCalled = true;
```

Comment:
Do we need to deal with booting at all? RestApplication does not care how the artifacts are registered, that's why we have `BootMixin` that can be applied to an app (or not).

I am proposing to simplify the tests in this file and remove everything related to booting.

Reply:
removed.